pymeasure/experiment/workers.py

```diff
@@ -116,9 +120,43 @@ def emit(self, topic, record):
             pass  # No dumps defined
         if topic == 'results':
             self.recorder.handle(record)
+        elif topic == 'batch results':
+            if isinstance(record, pd.DataFrame):
+                log.error(
+                    'Support for DataFrames when emitting batch results is not available yet.')
+                self.stop()
+            elif self._is_dictionary_of_sequences(record):
+                lengths = [len(value) for value in record.values()]
+                if not all(length == lengths[0] for length in lengths):
+                    log.warning(
+                        'Potential data loss: not all data columns have the same length '
+                        '(check your emitted results)'
+                    )
```

Comment:
I do think it is indeed good to check on the length of the different array. I might be inclined to even be more harsh and give an error when they don´t match, since there should never be a reason that data is incomplete (and if it is, is there data missing at the beginning or the end of a column?)

(as a side note, the `range(lengths[0])` can be "dangerous" if they're not all the same length and if the first entry happens to be shorter than one of the other entries)

Reply:
Good catch, I now log it as an error and return prematurely which also avoids the issue with `range(lengths[0])`, see 47905468.